int32 baseStep
int32 mainStep
int32 secStep
int32 toolStep